### PR TITLE
Use FromIterator trait rather than custom method

### PR DESCRIPTION
**Pull request by max-sixty**
ref https://github.com/rust-ndarray/ndarray/pull/642#discussion_r296068930

Is this the right direction?

I'm only now understanding how the rust import system works; I initially didn't realize you'd need an additional `use` statement each time to use `from_iter`. Should we be using `.collect()` in the examples? Or adding the `std::iter::FromIterator;` in the `ndarray` standard prelude?

### CODE REVIEW

**Comment by LukeMathWalker** [comment]
It definitely looks like the right direction @max-sixty!

Adding `std::iter::FromIterator` to the prelude would indeed reduce the chance of seeing existing code break with the new version, but it would still be a breaking change.
Not everyone uses the prelude and we are still removing the old `from_iter` function, which I think is the right decision: keeping both would lead to confusing compiler errors, asking for disambiguation when `use std::iter::FromIterator` is in scope.

An important point to keep in mind when working on this PR: make sure to update [ndarray for NumPy users](https://github.com/rust-ndarray/ndarray/tree/master/src/doc/ndarray_for_numpy_users) to reflect the changes you are introducing!

**Comment by max-sixty** [comment]
OK great; I updated the code throughout. 

Apart from the docs, what else?

Do we need to wait for a minor version to merge the `from_iter` removal?

File: src/arraytraits.rs [hunk 1]
```diff
@@ -127,7 +152,9 @@ where
     where
         I: IntoIterator<Item = A>,
     {
-        ArrayBase::from_iter(iterable)
+        // TODO: can I put this on one line?
```
[the remark this answers is not in the document]
Comment (max-sixty) [reply src/arraytraits.rs]: Thanks!

File: src/slice.rs [hunk 2]
```diff
@@ -697,6 +697,7 @@ pub unsafe fn deref_raw_view_mut_into_view_mut_with_life<'a, A, D: Dimension>(
 ///
 /// use ndarray::multislice;
 /// use ndarray::prelude::*;
+/// use std::iter::FromIterator;
 ///
 /// # fn main() {
 /// let mut arr = Array1::from_iter(0..12);
```
[the remark this answers is not in the document]
Comment (max-sixty) [reply src/slice.rs]: I tried to do something like this, but couldn't manage to convert a range to an iterator. Here's the error I get on this change (not that helpful in this case):

```rust

---- src/slice.rs - multislice (line 695) stdout ----
error[E0596]: cannot borrow `arr` as mutable, as it is not declared as mutable
  --> src/slice.rs:704:20
   |
9  | let arr: Array1<_> = (0..12).collect();
   |     --- help: consider changing this to be mutable: `mut arr`
10 | let (a, b, c, d) = multislice!(arr, [0..5], mut [6..;2], [1..6], mut [7..;2]);
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot borrow as mutable
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

**Comment by max-sixty** [comment]
Great, thanks for the thoughts.

Updated on comments, lmk any final changes!

**Comment by max-sixty** [comment]
Thanks @jturner314 !

**Comment by max-sixty** [comment]
Merge conflicts resolved

**Pull request merged**